org-code-javabuilder/theater/src/main/java/org/code/media/Color.java

```diff
@@ -0,0 +1,110 @@
+package org.code.media;
+
+public class Color {
+
+    /**
+     * Creates a color from a string representation.
+     * 
+     * @param color the string name of the color
+     * @throws IllegalArgumentException if the value specifies an unsupported color
+     *                                  name or illegal hexadecimal value
+     */
+    public Color(String color) throws IllegalArgumentException {
+
+    }
+
+    /**
+     * Create a new color based on the red, green, and blue values provided.
+     * 
+     * @param red   the red value from 0 - 255
+     * @param green the green value from 0 - 255
+     * @param blue  the blue value from 0 - 255
+     */
+    public Color(int red, int green, int blue) {
+
+    }
+
+    /**
+     * Returns the amount of red (ranging from 0 to 255).
+     * 
+     * @return a number representing the red value (between 0 and 255)
+     */
+    public int getRed() {
+    }
+
+    /**
+     * Returns the amount of green (ranging from 0 to 255).
+     * 
+     * @return a number representing the green value (between 0 and 255) of the
+     *         pixel.
+     */
+    public int getGreen() {
+    }
+
+    /**
+     * Returns the amount of blue (ranging from 0 to 255).
+     * 
+     * @return a number representing the blue value (between 0 and 255)
+     */
+    public int getBlue() {
+    }
+
+    /**
+     * Sets the amount of red (ranging from 0 to 255). Values below 0 will be
+     * ignored and set to 0, and values above 255 with be ignored and set to 255.
+     * 
+     * @param value the amount of red (ranging from 0 to 255) in the color of the
+     *              pixel.
+     */
+    public void setRed(int value) {
+    }
+
+    /**
+     * Sets the amount of green (ranging from 0 to 255). Values below 0 will be
+     * ignored and set to 0, and values above 255 with be ignored and set to 255.
+     * 
+     * @param value the amount of green (ranging from 0 to 255) in the color of the
+     *              pixel.
+     */
+    public void setGreen(int value) {
+    }
+
+    /**
+     * Sets the amount of blue (ranging from 0 to 255).
+     * 
+     * @param value the amount of blue (ranging from 0 to 255) in the color of the
+     *              pixel.
+     */
+    public void setBlue(int value) {
+    }
+
+    public static final Color white = new Color(255, 255, 255);
```

Comment:
I think the convention for Java constants is UPPER_SNAKE_CASE?

Reply:
`awt.Color` offers both so I chose lowercase: https://docs.oracle.com/javase/7/docs/api/java/awt/Color.html

That isn't to say it's the right choice :) I'll confer with curriculum, I think all caps can be a little off-putting but then again I used it for the instruments enum...

Reply:
I think the uppercase ones are the "correct" ones: https://stackoverflow.com/questions/15258824/how-come-java-has-two-color-values-for-each-color.

Reply:
I quickly checked and curriculum is okay with uppercase, so Molly has a change already for that :)